Syndemics/R/internalTools_crc.R

```diff
@@ -0,0 +1,122 @@
+#' Formula Objects
+#' Generic test of an object being interpretable as a formula
+#'
+#' @param x object to be tested
+#'
+#' @export
+
+is.formula <- function(x){
+  inherits(x, "formula")
+}
+
+#' Generates All Possible Combination of Interaction Terms
+#' Function that returns a list of every possible combination of interaction terms
+#'
+#' @param y Column: Column on the estimation side of the equation.
+#' @param x List: Columns to return every combination of
+#' @importFrom utils combn
+#' @importFrom stats as.formula
+#'
+#' @export
+
+formula_list <- function(y, x) {
+  n <- length(x)
+  all_formulas <- list()
+
+  # Individual terms
+  for (i in 1:n) {
+    all_formulas <- c(all_formulas, paste0(y, "~", x[i]))
+  }
+
+  # Additive and interaction terms
+  for (i in 2:n) {
+    combinations <- combn(x, i)
+    for (j in 1:ncol(combinations)) {
+      combination <- combinations[,j]
+
+      # Additive terms
+      all_formulas <- c(all_formulas, paste0(y, "~", paste(combination, collapse="+")))
+
+      # Interaction term
+      interaction_formula <- paste(combination, collapse="*")
+      all_formulas <- c(all_formulas, paste0(y, "~", interaction_formula))
+
+      # Additive terms with interaction term
+      for (k in 1:(i-1)) {
+        combinations_additive <- combn(combination, k)
+        for (l in 1:ncol(combinations_additive)) {
+          combination_additive <- combinations_additive[,l]
+          all_formulas <- c(all_formulas, paste0(y, "~", paste(combination_additive, collapse="+"), "+", interaction_formula))
+        }
+      }
+    }
+  }
+
+  return(lapply(unique(all_formulas), as.formula))
+}
+
+#' Internal function to create a formula based on correlation among databases listed in \code{CRC}
+#'
+#' @param corr_matrix A correlation matrix
+#' @param threshold Threshold for interacting two databases
+#' @param freq.column frequency column, e.g. N_ID
+#'
+#' @keywords internal
+formula_corr <- function(corr_matrix, threshold, freq.column){
+  corr_upper <- corr_matrix; corr_upper[upper.tri(corr_matrix, diag = TRUE)] <- NA
```

Comment:
Any particular reason this is on one line? It is a bit tricky to read

Reply:
I was probably thinking that the corr_matrix 'setup' would be easiest to denote on one line - this has been changed

Reply:
Resolved: [a2d2dae](https://github.com/SyndemicsLab/Syndemics/pull/8/commits/a2d2daefb720df55019d2c12d612298b82c16393)